src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMFileSystem.java

```diff
@@ -84,8 +84,11 @@ protected GitHubSCMFileSystem(GitHub gitHub, GHRepository repo, String refName,
         if (rev != null) {
             if (rev.getHead() instanceof PullRequestSCMHead) {
                 PullRequestSCMHead pr = (PullRequestSCMHead) rev.getHead();
-                assert !pr.isMerge(); // TODO see below
-                this.ref = ((PullRequestSCMRevision) rev).getPullHash();
+                if (pr.isMerge()) {
+                    this.ref = ((PullRequestSCMRevision) rev).getMergeHash();
```

Comment:
This will in general produce different results from what `GitHubSCMBuilder.build` does when you actually `checkout scm`.

Reply:
Will it?  Git merge is in one place isn't git merge in another?  
Could you explain or point me to documentation that will help me understand your point?  

The `merge_commit_sha` is guaranteed to be be a valid automatic merge. 
How significant can the differences be?